File: 0633.平方数之和/0633-平方数之和.c
bool judgeSquareSum(int c){
    if (c < 0) {
        return false;
    }
    long long int left = 0;
    long long int right =  (sqrt(c) + 1);
    
    while (left <= right) {
        int val = left*left + right*right;
        if (val == c) {
            return true;
        } else if (val > c) {
            right -= 1;
        } else {
            left += 1;
        }
    }
    
    return false;
}

File: 0633.平方数之和/0633-平方数之和.rs
impl Solution {
    pub fn judge_square_sum(c: i32) -> bool {
        if c < 0 {
            return false;
        }
        
        let mut left = 0usize;
        let mut right = ((c as f32).sqrt() + 1.0) as usize;
        let c = c as usize;
        while left <= right {
            let val = left*left + right*right;
            if val == c {
                return true;
            } else if val < c {
                left += 1;
            } else {
                right -= 1;
            }
        }
        return false;
    }
}